packages/main/src/plugin/index.spec.ts

```diff
@@ -413,3 +419,104 @@ test('ipcMain.handle returns caught error as objects message property if it is n
   const handleReturn = await handle(undefined, '1');
   expect(handleReturn.error).toEqual({ message: nonErrorInstance });
 });
+
+describe.each<{
+  handler: string;
+  methodName: 'createContainerProviderConnection' | 'createKubernetesProviderConnection';
+}>([
+  {
+    handler: 'provider-registry:createContainerProviderConnection',
+    methodName: 'createContainerProviderConnection',
+  },
+  {
+    handler: 'provider-registry:createKubernetesProviderConnection',
+    methodName: 'createKubernetesProviderConnection',
+  },
+])('$handler', async ({ handler, methodName }) => {
+  test('createTask is called', async () => {
+    vi.spyOn(TaskManager.prototype, 'createTask');
+    vi.spyOn(NavigationManager.prototype, 'navigateToProviderTask');
+    vi.spyOn(ProviderRegistry.prototype, 'getProviderInfo').mockReturnValue({
+      name: 'provider1',
+    } as ProviderInfo);
+    await pluginSystem.initExtensions(new Emitter<ConfigurationRegistry>());
+    const handle = handlers.get(handler);
+    expect(handle).not.equal(undefined);
+    await handle(undefined, 'internal1', { key1: 'value1', key2: 42 }, 'logger1', 'token1', 'task1');
+    expect(TaskManager.prototype.createTask).toHaveBeenCalledOnce();
+    const params = vi.mocked(TaskManager.prototype.createTask).mock.calls[0]?.[0];
+    if (!params) {
+      // this is already expected
+      throw new Error('param should be defined');
+    }
+    expect(params.title).toEqual(`Creating provider1 provider`);
+    expect(params.action?.name).toEqual(`Open task`);
+    const execute = params.action?.execute;
+    expect(execute).toBeDefined();
+    if (!execute) {
+      throw new Error('execute should be defined');
+    }
+    execute(new TaskImpl('task1id', 'task1name'));
+    expect(NavigationManager.prototype.navigateToProviderTask).toHaveBeenCalledOnce();
+    expect(NavigationManager.prototype.navigateToProviderTask).toHaveBeenCalledWith('internal1', 'task1');
```

Comment:
Those tests are very dense, pretty hard to read

Reply:
added some comments